index.js

```diff
@@ -2,32 +2,42 @@
 
 /* eslint-disable no-underscore-dangle */
 class NotificationBase {
-
     /**
      * Constructor for Build Notification
      * @method constructor
      * @param {Object}      config              Configuration specific to the plugin
-     * @param {Object}      server              API server
-     * @param {String}      eventName           Event name the notification plugin listen to
      * @return {NotificationBase}
      */
-    constructor(config, server, eventName) {
+    constructor(config) {
         this.config = config;
-        this.server = server;
-        this.eventName = eventName;
+    }
+
+    /**
+     * Get event list for receiving Screwdriver API events
+     * @method events
+     * @return {Array}
+     */
+    get events() {
+        return ['build_status'];
     }
 
     /**
      * Listen to the event and notify users based on their job settings
+     * @param {Object}      buildData           Build data emitted with some event from Screwdriver
      * @method notify
-     * @return {Promise}
```

Comment:
Interesting choice. Promises have the ability to catch any errors that are thrown, similar to a try-catch block. I guess I don't understand why we would want to move away from a Promise-based function call.